### ISSUE TITLE
false postive SC2329

### ISSUE THREAD

**Issue by dotysan** [koalaman/shellcheck#3385]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2329
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /usr/bin/env bash
#
XTRACE=yep

main() {
    echo "hello:" "$@"
    return 1
}

if [[ "${XTRACE:-}" ]]
then
    PS4func() {
        printf "%b" "+${BASH_LINENO[0]}: "
          
    }
    PS4='\r$(PS4func)'
    set -o xtrace
fi

main "$@"
exit $?
```

#### Here's what shellcheck currently says:
```
Line 12	SC2329: This function is never invoked. Check usage (or ignored if invoked indirectly).
```
#### Here's what I wanted or expected to see:

That function is evoked and the script runs fine.

The exit $? on the last line seems to confuse SC2329.
